sdk/nodejs/x/automation/localWorkspace.ts

```diff
@@ -0,0 +1,357 @@
+            promises.push(this.removeConfig(stackName, key));
+        }
+        return Promise.resolve(<any>{});
+    }
+    async refreshConfig(stackName: string): Promise<ConfigMap> {
+        await this.selectStack(stackName);
+        await this.runPulumiCmd(["config", "refresh", "--force"]);
+        return this.getAllConfig(stackName);
+    }
+    getEnvVars(): { [key: string]: string } {
+        return this.envVars;
+    }
+    setEnvVars(envs: { [key: string]: string }): void {
+        this.envVars = { ...this.envVars, ...envs };
+    }
+    setEnvVar(key: string, value: string): void {
+        this.envVars[key] = value;
+    }
+    unsetEnvVar(key: string): void {
+        delete this.envVars[key];
+    }
+    getWorkDir(): string {
+        return this.workDir;
+    }
+    getPulumiHome(): string | undefined {
+        return this.pulumiHome;
+    }
+    async whoAmI(): Promise<string> {
+        const result = await this.runPulumiCmd(["whoami"]);
+        return Promise.resolve(result.stdout.trim());
+    }
+    async stack(): Promise<StackSummary | undefined> {
+        const stacks = await this.listStacks();
+        for (const stack of stacks) {
+            if (stack.current) {
+                return Promise.resolve(stack);
+            }
+        }
+        return Promise.resolve(undefined);
+    }
+    async listStacks(): Promise<StackSummary[]> {
+        const result = await this.runPulumiCmd(["stack", "ls", "--json"]);
+        const stacks: StackSummary[] = JSON.parse(result.stdout);
+        return Promise.resolve(stacks);
+    }
+    getProgram(): PulumiFn | undefined {
+        return this.program;
+    }
+    setProgram(program: PulumiFn): void {
+        this.program = program;
+    }
+    serializeArgsForOp(_: string): Promise<string[]> {
+        // LocalWorkspace does not take advantage of this extensibility point.
+        return Promise.resolve([]);
+    }
+    postCommandCallback(_: string): Promise<void> {
+        // LocalWorkspace does not take advantage of this extensibility point.
+        return Promise.resolve();
+    }
+    private async runPulumiCmd(
+        args: string[],
+    ): Promise<CommandResult> {
+        let envs: { [key: string]: string } = {};
+        if (this.pulumiHome) {
+            envs["PULUMI_HOME"] = this.pulumiHome;
+        }
+        envs = { ...envs, ...this.getEnvVars() };
+        return runPulumiCmd(args, this.workDir, envs);
+    }
+}
+
+export type LocalWorkspaceOpts = {
+    workDir?: string,
+    pulumiHome?: string,
+    program?: PulumiFn,
+    envVars?: { [key: string]: string },
+    secretsProvider?: string,
+    projectSettings?: ProjectSettings,
+    stackSettings?: { [key: string]: StackSettings },
+};
+
+export const settingsExtensions = [".yaml", ".yml", ".json"];
+
+const getStackSettingsName = (name: string): string => {
+    const parts = name.split("/");
+    if (parts.length < 1) {
+        return name;
+    }
+    return parts[parts.length - 1];
+};
```

Comment:
It'll be internal (which it currently is if I'm not mistaken?). What's the guidance on using `function` vs the arrow form for things like this? Is this just a stylistic thing?

Reply:
> (which it currently is if I'm not mistaken?)

Yes, it is internal (it's not exported), I as just double checking that it would remain internal.

> What's the guidance on using `function` vs the arrow form for things like this? Is this just a stylistic thing?

For things like this, it's mostly a style thing. We've largely used `function` in most places like this elsewhere, so would be nice to continue to do that for consistency, but, my personal care level is pretty low here. More of a nitpick than anything else.

Reply:
Fixed these for consistency.